sql/cagg_api.sql

```diff
@@ -0,0 +1,182 @@
+              FROM _timescaledb_catalog.continuous_aggs_hypertable_invalidation_log
+             WHERE hypertable_id = l_hypertable_id
+         ),
+         -- Since a range can end at the end of a bucket and this
+         -- range should not include the next bucket, we need to
+         -- subtract one microsecond before computing the bucket and
+         -- then add the width again.
+         ranges AS MATERIALIZED (
+           SELECT width,
+                  tstzrange(@extschema@.time_bucket(width, start),
+                            @extschema@.time_bucket(width, finish - '1 microsecond'::interval) + width) AS bucket
+             FROM timestamps CROSS JOIN UNNEST(bucket_widths) w(width)
+         )
+      SELECT width, pg_current_snapshot()::text, range_agg(bucket) ranges
+        FROM ranges GROUP BY width
+   );
+END
+$body$
+LANGUAGE plpgsql
+SET search_path = pg_catalog, pg_temp;
+
+CREATE OR REPLACE FUNCTION _timescaledb_functions.get_hypertable_invalidations(
+   hypertable REGCLASS,
+   base TIMESTAMP,
+   bucket_widths INTERVAL[]
+) RETURNS TABLE (bucket_width INTERVAL, token TEXT, invalidations TSMULTIRANGE) AS
+$body$
+DECLARE
+   l_hypertable_id INTEGER := _timescaledb_functions.get_hypertable_id(hypertable, 'timestamp'::regtype);
+BEGIN
+   RETURN QUERY (
+      WITH
+         -- Collect ranges from the invalidation log and convert them
+         -- to correct type.
+         timestamps AS MATERIALIZED (
+            SELECT _timescaledb_functions.to_timestamp_without_timezone(lowest_modified_value) AS start,
+                   _timescaledb_functions.to_timestamp_without_timezone(greatest_modified_value) AS finish
+              FROM _timescaledb_catalog.continuous_aggs_hypertable_invalidation_log
+             WHERE hypertable_id = l_hypertable_id
+         ),
+         -- Compute bucket-aligned ranges from the ranges above. Since
+         -- a range can end at the end of a bucket and this range
+         -- should not include the next bucket, we need to subtract
+         -- one microsecond before computing the bucket and then add
+         -- the width again.
+         ranges AS MATERIALIZED (
+           SELECT width,
+                  tsrange(@extschema@.time_bucket(width, start),
+                          @extschema@.time_bucket(width, finish - '1 microsecond'::interval) + width) AS bucket
+             FROM timestamps CROSS JOIN UNNEST(bucket_widths) w(width)
+         )
+      SELECT width, pg_current_snapshot()::text, range_agg(bucket)
+        FROM ranges GROUP BY width
+   );
+END
+$body$
+LANGUAGE plpgsql
+SET search_path TO pg_catalog, pg_temp;
+
+-- Accept a set of hypertable invalidations for a hypertable.
+--
+-- This procedure is used to accept all invalidations for a hypertable
+-- using the token returned by a previous call of
+-- get_hypertable_invalidations().
+CREATE OR REPLACE PROCEDURE _timescaledb_functions.accept_hypertable_invalidations(
```

Comment:
This name still confusing but don't have a better idea tought. Since it is an internal API we can change it later.

Reply:
It is based on the names of the logical decoding functions. Wanted to avoid "commit" since this is not transactional.